src/common/task_system/task_scheduler.cpp

```diff
@@ -81,20 +85,25 @@ void TaskScheduler::removeErroringTask(uint64_t scheduledTaskID) {
     for (auto it = taskQueue.begin(); it != taskQueue.end(); ++it) {
         if (scheduledTaskID == (*it)->ID) {
             taskQueue.erase(it);
+            cv.notify_all();
             return;
         }
     }
 }
 
 void TaskScheduler::runWorkerThread() {
+    std::unique_lock<std::mutex> lck{mtx, std::defer_lock};
     while (true) {
-        if (stopThreads.load()) {
-            break;
+        lck.lock();
+        while (taskQueue.empty() && !stopThreads) {
```

Comment:
Can the while loop be simplified as the following line of code?
```cpp
        cv.wait(lck, [&] { return !taskQueue.empty() || stopThreads; });
```

Reply:
Done